src/lib.rs

```diff
@@ -1250,22 +1274,34 @@ impl<T> fmt::Display for TrySendError<T> {
 }
 
 /// An error returned from [`Receiver::recv()`].
-///
-/// Received because the channel is empty and closed.
 #[derive(PartialEq, Eq, Clone, Copy, Debug)]
-pub struct RecvError;
+pub enum RecvError {
+    /// The channel has overflowed since the last element was seen.  Future recv operations will
+    /// succeed, but some messages have been skipped.
+    Overflowed,
+
+    /// The channel is empty and closed.
+    Closed,
+}
 
 impl error::Error for RecvError {}
 
 impl fmt::Display for RecvError {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "receiving from an empty and closed channel")
+        match self {
+            Self::Overflowed => write!(f, "receiving skipped some messages"),
+            Self::Closed => write!(f, "receiving from an empty and closed channel"),
+        }
     }
 }
 
 /// An error returned from [`Receiver::try_recv()`].
 #[derive(PartialEq, Eq, Clone, Copy, Debug)]
 pub enum TryRecvError {
+    /// The channel has overflowed since the last element was seen.  Future recv operations will
+    /// succeed, but some messages have been skipped.
+    Overflowed,
```

Comment:
It could be interesting if this was:
```suggestion
    Overflowed(pub usize),
```
and would keep track how many messages have been dropped since it last succeeded.

Reply:
Pretty simple to do.